libs/ha/ha_common_libs_k8s.py

```diff
@@ -1855,3 +1856,87 @@ def dnld_obj_verify_chcksm(self, s3_test_obj, bucket, obj, download_path, upload
             return False, download_checksum
         LOGGER.info("Successfully downloaded the object and verified the checksum")
         return True, download_checksum
+
+    def object_overwrite_dnld(self, s3_test_obj, s3_data, iteration, random_size, event=None,
+                              queue=None, background=False):
+        """
+        Function to create bucket, put object and overwrite existing object
+        :param s3_test_obj: Object of s3 test lib
+        :param s3_data: Dict that contains bucket and object info
+        ({self.bucket_name: [self.object_name, file_size]})
+        :param iteration: Number of iterations
+        :param random_size: Flag to select random size of object
+        :param event: Event for background IOs
+        :param queue: Queue to fill output in case of background IOs
+        :param background: Flag to start background process
+        :return: (bool, dict)
+        """
+        checksums = dict()
+        fail_count = 0
+        test_dir_path = os.path.join(TEST_DATA_FOLDER, "HAOverWrite")
+        if not os.path.isdir(test_dir_path):
+            LOGGER.debug("Creating file path %s", test_dir_path)
+            system_utils.make_dirs(test_dir_path)
+
+        resp = s3_test_obj.bucket_list()[1]
+        for bkt, value in s3_data.items():
+            object_name = value[0]
+            object_size = value[1]
+            if bkt not in resp:
+                LOGGER.info("Creating a bucket with name %s and uploading object of size %s MB",
+                            bkt, object_size)
+                file_path = os.path.join(test_dir_path, f"{bkt}.txt")
+                _ = s3_test_obj.create_bucket_put_object(bkt, object_name, file_path, object_size)
+                LOGGER.info("Created a bucket with name %s and uploaded object %s of size %s MB",
+                            bkt, object_name, object_size)
+                system_utils.remove_file(file_path)
```

Comment:
why remove file_path here?

Reply:
If file size is large, it will create memory issues. So removing file path here.